src/main/java/domain/TodoRepository.java

```diff
@@ -26,4 +26,10 @@ public void deleteTodo(final long todoId) {
         Todo todo = em.find(Todo.class, todoId);
         em.remove(todo);
     }
+
+    @Transactional
```

Comment:
Bitte in der Resource verwenden. Sonst ist nur die updateTodo hier transactional. Schlägt es aber zwischendrin fehl, dann gibt es kein Rollback.